### PR TITLE
JAMES-2077 The RFC 1123 time formatter doesn't support RFC 5322 year …

### PR DESCRIPTION
**Pull request by aduprat**
…definition

### CODE REVIEW

File: server/container/util-java8/src/main/java/org/apache/james/util/date/ImapDateTimeFormatter.java [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+import java.util.Map;
+
+public class ImapDateTimeFormatter {
+
+    private static final int INITIAL_YEAR = 1970;
+
+    public static final DateTimeFormatter RFC_5322_DATE_TIME;
+
+    static {
+        Map<Long, String> dow = new HashMap<>();
```
Comment (chibenwa) [inline server/container/util-java8/src/main/java/org/apache/james/util/date/ImapDateTimeFormatter.java]: Can we find a more explicit name than dow?

Comment (chibenwa) [inline server/container/util-java8/src/main/java/org/apache/james/util/date/ImapDateTimeFormatter.java]: ImmutableMap ?

File: server/container/util-java8/src/main/java/org/apache/james/util/date/ImapDateTimeFormatter.java [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+        dow.put(1L, "Mon");
+        dow.put(2L, "Tue");
+        dow.put(3L, "Wed");
+        dow.put(4L, "Thu");
+        dow.put(5L, "Fri");
+        dow.put(6L, "Sat");
+        dow.put(7L, "Sun");
+        Map<Long, String> moy = new HashMap<>();
```
Comment (chibenwa) [inline server/container/util-java8/src/main/java/org/apache/james/util/date/ImapDateTimeFormatter.java]: Idem here...

File: server/container/util-java8/src/main/java/org/apache/james/util/date/ImapDateTimeFormatter.java [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+import java.util.HashMap;
+import java.util.Map;
+
+public class ImapDateTimeFormatter {
+
+    private static final int INITIAL_YEAR = 1970;
+
+    public static final DateTimeFormatter RFC_5322_DATE_TIME;
+
+    static {
```
Comment (chibenwa) [inline server/container/util-java8/src/main/java/org/apache/james/util/date/ImapDateTimeFormatter.java]: I don't like static blocks. Could you please replace it by a static method?

File: server/container/util-java8/src/main/java/org/apache/james/util/date/ImapDateTimeFormatter.java [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+        moy.put(7L, "Jul");
+        moy.put(8L, "Aug");
+        moy.put(9L, "Sep");
+        moy.put(10L, "Oct");
+        moy.put(11L, "Nov");
+        moy.put(12L, "Dec");
+        RFC_5322_DATE_TIME = new DateTimeFormatterBuilder()
+                .parseCaseInsensitive()
+                .parseLenient()
+                .optionalStart()
```
Comment (mbaechler) [inline server/container/util-java8/src/main/java/org/apache/james/util/date/ImapDateTimeFormatter.java]: it lacks indent

File: server/container/util-java8/src/main/java/org/apache/james/util/date/ImapDateTimeFormatter.java (hunk 2 above)
Comment (mbaechler) [inline server/container/util-java8/src/main/java/org/apache/james/util/date/ImapDateTimeFormatter.java]: static mutable structures ? bad idea

**Comment by aduprat** [comment]
I will add some unit test on the new DateTimeFormatter

File: mailbox/elasticsearch/src/test/java/org/apache/james/mailbox/elasticsearch/json/HeaderCollectionTest.java [hunk 5]
```diff
@@ -223,6 +223,16 @@ public void dateShouldBeRetreived() {
     }
 
     @Test
+    public void centuryShouldBeAppenToDateWhenNone() {
```
Comment (fvn-linagora) [inline mailbox/elasticsearch/src/test/java/org/apache/james/mailbox/elasticsearch/json/HeaderCollectionTest.java]: `shouldAppendCenturyWhenMissing` or `partialYearShouldBeCompleted` ?

**Comment by chibenwa** [comment]
I merge the fix as is. If additional comments needs to be done, we can handle them later.

**Comment by chibenwa** [comment]
Merged

**Pull request closed without merging**